addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -195,11 +195,16 @@ export class SelfOrder extends Reactive {
     }
 
     get currentOrder() {
-        if (this.editedOrder && this.editedOrder.state === "draft") {
+        if (
+            this.editedOrder &&
+            (this.editedOrder.state === "draft" ||
+                (this.editedOrder.state === "paid" && this.editedOrder.amount_total === 0))
+        ) {
             return this.editedOrder;
         }
-
-        const existingOrder = this.orders.find((o) => o.state === "draft");
+        const existingOrder = this.orders.find(
+            (o) => o.state === "draft" || (o.state === "paid" && o.amount_total === 0)
+        );
```

Comment:
Nice try, but you can't do it that way.

Unfortunately, self order code covers a large number of behaviors, and this modification breaks at least one of them.

Step to reproduce:
- Install pos_self_order.
- Set up a restaurant with the code self order pay after each order.
- Create an order with a zero amount. Pay it.
- Try to do it again and an error occurs.

The problem:
In pay after each order, in the mobile self order, when you pay an order with an amount of zero, the server returns it directly in the paid state, which means that when you want to place another order, the `currentOrder` doesn't return a new order but the one that has been paid.

Image:
![image](https://github.com/odoo/odoo/assets/63289800/3070f520-5fcd-4230-810e-d56603058e1f)

![image](https://github.com/odoo/odoo/assets/63289800/451b44d9-0779-4250-9578-52cc67a3fb54)